### ISSUE TITLE
Internal Server Error when assign DS to Federations when Federations doesn't exist

### ISSUE THREAD

**Issue by dpham692** [apache/trafficcontrol#4090]
## I'm submitting a ...
- [x] bug report
- [ ] new feature / enhancement request
- [ ] improvement request (usability, performance, tech debt, etc.)
- [ ] other 

## Traffic Control components affected ...
- [ ] CDN in a Box
- [ ] Documentation
- [ ] Grove
- [ ] Traffic Control Client
- [ ] Traffic Monitor
- [x] Traffic Ops
- [ ] Traffic Ops ORT
- [ ] Traffic Portal
- [ ] Traffic Router
- [ ] Traffic Stats
- [ ] Traffic Vault
- [ ] unknown

## Current behavior:
Do not create Federations. Then assign Delivery Service to Federations
POST Request : `https://{{TO_BASE_URL}}/api/{{api_version}}/federations/1000/deliveryservices`
With this json payload 
```json
{
        "dsIds": [2],
        "replace": true
}
```
Status 500 Internal Server Error
```json
{
    "alerts": [
        {
            "text": "Internal Server Error",
            "level": "error"
        }
    ]
}
```
## Expected / new behavior:
Return 404

**Comment by ocket8888** [comment]
Should also have appropriate error alert.

**Comment by mhoppa** [comment]
This was fixed in the linked PR unsure why this wasnt closed @ocket8888 @rawlinp @mitchell852 can one of you guys close this?

**Comment by ocket8888** [comment]
Because of your wording:
> _"This PR fixes #4087 & #4090"_
will close 4087 but just creates a link to 4090. What you wanted to do was:
> _"This PR  fixes #4087 and fixes #4090"_
which will close them both

**Comment by mhoppa** [comment]
ahh TIL
